lesson_5_hw.js

```diff
@@ -0,0 +1,93 @@
+/*
+1. 
+  Напиши функцію map(fn, array), яка приймає на вхід функцію та масив, 
+  та обробляє кожен елемент масиву цією функцією, повертаючи новий масив. 
+  Приклад:
+*/
+
+function map(fn, array) {
+  //creating a new array
+  let result = [];
+
+  //iterating through the array and applying the function
+  for (number of array) {
+    result.push(fn(number));
+  }
+  return result;
+}
+function square(x) {
+  return x * x;
+} // повертає квадрат числа
+
+console.log(map(square, [1, 2, 3, 4])); // [1, 4, 9, 16]
+console.log(map(square, [])); // []
+
+// Зверніть увагу: функція не повинна змінювати переданий їй масив:
+
+var arr = [1, 2, 3];
+console.log(map(square, arr)); // [1, 4, 9]
+console.log(arr); // [1, 2, 3]
+
+/*
+  2. Написати функцію яка форматує текст: 
+    - прибирає пробіли з початку і кінця, 
+    - прибарає всі займі пробіли (два і більше підряд)
+    - кожне речення починає з великої букви, а всі наступні символи в речені робить маленькими
+*/
+function formatText(text) {
+  // cleaning up extra spaces and formatting entire text to the lowercase
+  let formattedText = text.trim().replaceAll("  ", "").toLowerCase();
+
+  // splitting text into sentences
+  formattedText = formattedText.split(". ");
+
+  // capitalizing first letter of each sentence
+  for (let sentence of formattedText) {
+    sentence = sentence.charAt(0).toUpperCase() + sentence.slice(1);
```

Comment:
```suggestion
    newArr = sentence.charAt(0).toUpperCase() + sentence.slice(1);
```
you do not change original value in the array "formattedText" here, might be useful to create a new array and push formatted sentence there 

BTW, letters are not capitalized in the log:

Reply:
<img width="330" alt="image" src="https://github.com/bohdanhulobov/Hillel_js_aqa/assets/30954131/ce300d6e-9c86-4e93-b73e-d93341bee1f7">

Reply:
@Art-auto fixed